### ISSUE TITLE
J9MONITORTABLE_OBJECT_LOAD macros have unnecessary duplicate check

### ISSUE THREAD

**Issue by DanHeidinga** [eclipse/openj9#6200]
There are duplicate checks in these macros that can be made more efficient:
https://github.com/eclipse/openj9/blob/2673d23e5c7525b80254356df1635565f1642ef0/runtime/oti/j9accessbarrier.h#L600-L612

I don't recall if there was a reason to have the extra check in these macros.

ie:
```c
((J9_GC_READ_BARRIER_TYPE_RANGE_CHECK == (vmThread)->javaVM->gcReadBarrierType) ? \ 
 ((j9object_t)J9VMTHREAD_JAVAVM(vmThread)->memoryManagerFunctions->j9gc_objaccess_monitorTableReadObject((vmThread), (j9object_t *)(objectSlot))) : \ 
 ((j9object_t)J9VMTHREAD_JAVAVM(vmThread)->memoryManagerFunctions->j9gc_objaccess_monitorTableReadObject((vmThread), (j9object_t *)(objectSlot))))) 
  ```

**Comment by DanHeidinga** [comment]
@amicic @dmitripivkine Do you recall if there was a reason for the second check?  The macros could be simplified to:

```c
#define J9MONITORTABLE_OBJECT_LOAD(vmThread, objectSlot) \ 
 		((J9_GC_READ_BARRIER_TYPE_NONE == (vmThread)->javaVM->gcReadBarrierType) ? \ 
 		(*(j9object_t *)(objectSlot)) : \ 
 		((j9object_t)J9VMTHREAD_JAVAVM(vmThread)->memoryManagerFunctions->j9gc_objaccess_monitorTableReadObject((vmThread), (j9object_t *)(objectSlot))))) 
  
```

**Comment by amicic** [comment]
You are racing with this PR :)
https://github.com/eclipse/openj9/pull/6164#discussion_r295016207

**Comment by DanHeidinga** [comment]
Closing this as it's being addressed in #6164 (comment)
